src/catch2/internal/catch_test_case_registry_impl.cpp

```diff
@@ -79,11 +55,7 @@ namespace {
             std::sort( indexed_tests.begin(),
                        indexed_tests.end(),
                        []( TestWithHash const& lhs, TestWithHash const& rhs ) {
-                           if ( lhs.first == rhs.first ) {
-                               return lhs.second.getTestCaseInfo() <
-                                      rhs.second.getTestCaseInfo();
-                           }
-                           return lhs.first < rhs.first;
+                              return lhs.first <= rhs.first;
```

Comment:
This is wrong, the original code was correct -> the hashes can collide even if the inputs are different, so we want to use the `TestCaseInfo` comparison as the tiebreaker to ensure consistent order.